accumulator_updater/programs/mock-cpi-caller/src/instructions/add_price.rs

```diff
@@ -0,0 +1,129 @@
+use {
+    crate::{
+        instructions::sighash,
+        message::{
+            get_schemas,
+            price::{
+                CompactPriceMessage,
+                FullPriceMessage,
+            },
+            AccumulatorSerializer,
+        },
+        state::{
+            PriceAccount,
+            PythAccountType,
+        },
+    },
+    accumulator_updater::program::AccumulatorUpdater as AccumulatorUpdaterProgram,
+    anchor_lang::{
+        prelude::*,
+        solana_program::sysvar,
+    },
+};
+
+pub fn add_price<'info>(
+    ctx: Context<'_, '_, '_, 'info, AddPrice<'info>>,
+    params: AddPriceParams,
+) -> Result<()> {
+    let mut account_data: Vec<Vec<u8>> = vec![];
+    let schemas = get_schemas(PythAccountType::Price);
+
+    {
+        let pyth_price_acct = &mut ctx.accounts.pyth_price_account.load_init()?;
```

Comment:
I don't think you have to use load_init, the account got initialized in the context already

Reply:
for PDAs that use  `zero-copy`/`AccountLoader` it still needs to be called with `load_init()` otherwise the discriminator isn't set. 
This part of the code isn't that important though since it's just mocking pyth price accounts

I used the anchor test code for `CreateBar` from here as a template - https://github.com/coral-xyz/anchor/blob/master/tests/zero-copy/programs/zero-copy/src/lib.rs#L95